threedi_api_qgis_client/widgets/wizard.py

```diff
@@ -589,30 +990,85 @@ def run_new_simulation(self):
-                                               organisation=organisation_uuid, duration=duration)
-            sim_id = new_simulation.id
-            ptype, poffset, pduration, punits, pvalues = self.p3.main_widget.get_precipitation_data()
-            if ptype == CONSTANT_RAIN:
-                tc.add_constant_precipitation(sim_id, value=pvalues, units=punits, duration=pduration, offset=poffset)
-            elif ptype == CUSTOM_RAIN or ptype == DESIGN_RAIN:
-                tc.add_custom_precipitation(sim_id, values=pvalues, units=punits, duration=pduration, offset=poffset)
-            tc.make_action_on_simulation(sim_id, name='queue')
-            current_status = tc.simulation_current_status(sim_id)
-            self.new_simulation = new_simulation
-            self.new_simulation_status = current_status
-            msg = f"Simulation {new_simulation.name} added to queue!"
-            self.parent_dock.communication.bar_info(msg, log_text_color=QColor(Qt.darkGreen))
+            simulation_templates = dict()
+            self.new_simulations = []
+            self.new_simulation_statuses = {}
+            for simulation in range(1, (self.init_conditions.number_of_simulations + 1)):
+                ptype, poffset, pduration, punits, pvalues = None, None, None, None, None
+                breache_id, breache, width, d_duration = None, None, None, None
+                laterals = []
+                if hasattr(self, 'p3'):
+                    ptype, poffset, pduration, punits, pvalues = self.p3.main_widget.get_precipitation_data("Simulation" + str(simulation))
+                if hasattr(self, "breaches_page"):
+                    breache_id, breache, width, d_duration = self.breaches_page.main_widget.get_breaches_data("Simulation" + str(simulation))
+                if hasattr(self, "laterals_page"):
+                    laterals = self.laterals_page.main_widget.get_laterals_data()
+
+                if self.p4.main_widget.cb_save_template.isChecked():
+                    template_name = self.p4.main_widget.template_name.text()
+                    simulation_template = {}
+                    simulation_template["template_name"] = template_name
+                    simulation_template["name"] = name
+                    simulation_template["threedimodel"] = threedimodel_id
+                    simulation_template["organisation"] = organisation_uuid
+                    simulation_template["ptype"] = ptype
+                    simulation_template["poffset"] = poffset
+                    simulation_template["pduration"] = pduration
+                    simulation_template["punits"] = punits
+                    simulation_template["pvalues"] = pvalues
+                    simulation_templates[simulation] = simulation_template
+                    path = os.path.join(os.path.dirname(os.path.dirname(os.path.realpath(__file__))), "templates.json")
```

Comment:
This path is defined twice in the code. I think it should be putted into global variable like TEMPLATE_PATH or something like that.